pkg/k8sclient/kubeconfig.go

```diff
@@ -228,8 +228,19 @@ func newClientInfo(config *rest.Config) (*ClientInfo, error) {
 	if err != nil {
 		return nil, err
 	}
+	netClient, err := netclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
 
-	netclient, err := netclient.NewForConfig(config)
+	watchConfig := rest.CopyConfig(config)
+	// Do not set timeout for watches and delegate timeout to client-go
```

Comment:
why dont we want to do that for for netClient as well ?

that is set zero (or leave unset) at L215

Reply:
For non-watch calls, it might be a good practice to set a timeout. I saw similar behaviours with kube-scheduler:
```
kube-scheduler/v1.24.4 (linux/amd64) kubernetes/95ee5ab/leader-election -> get -> /apis/coordination.k8s.io/v1/namespaces/kube-system/leases/kube-scheduler?timeout=5s -> 200
```

The other reason I didn't update for the non-watch clients is for functional parity.

Reply:
ack, thx for the info.